lib/manager.js

```diff
@@ -381,10 +380,18 @@ module.exports = class Manager {
         });
     }
 
-    _render(compiled, context, request) {
+
+    async _render(compiled, context, request, callback) {
 
         if (this._context) {
-            let base = typeof this._context === 'function' ? this._context(request) : this._context;
+
+            let base = this._context;
+
+            if (typeof this._context === 'function') {
+
+                base = await this._context(request);
```

Comment:
Added benefit to prevent people from messing with `this`, unless that would be desired?
```suggestion
            if (typeof base === 'function') {

                base = await base(request);
```